ipatests/test_integration/test_client.py

```diff
@@ -0,0 +1,74 @@
+#
+# Copyright (C) 2019  FreeIPA Contributors see COPYING for license
+#
+import random
+
+from ipatests.test_integration.base import IntegrationTest
+from ipatests.pytest_ipa.integration import tasks
+
+
+class TestSSHFromClient(IntegrationTest):
+    """
+    Tests ssh from client when ldap_deref_threshold=0 is added to sssd.conf
+    Automated test for https://pagure.io/SSSD/sssd/issue/3979
+    """
+    num_clients = 1
+
+    @classmethod
+    def install(cls, mh):
+        tasks.install_master(cls.master)
+        cls.client = cls.clients[0]
+        tasks.install_client(cls.master, cls.client)
+
+    def test_ssh_from_client(self):
+        """
+        Test ssh from client after adding ldap_deref_threshold=0 to sssd.conf
+
+        Steps:
+        1. setup a master and a client
+        2. add ldap_deref_threshold=0 to sssd.conf on master
+        3. add an ipa user
+        4. ssh from client to master using the user created in step 3
+        """
+
+        username = "testuser" + str(random.randint(200000, 9999999))
+        # add ldap_deref_threshold=0 to /etc/sssd/sssd.conf
+        domain = self.master.domain
+        tasks.modify_sssd_conf(
+            self.master,
+            domain.name,
+            {
+                'ldap_deref_threshold': 0
+            },
+        )
+        self.master.run_command(['systemctl', 'restart', 'sssd.service'])
+
+        # kinit admin
+        tasks.kinit_admin(self.master)
+
+        # add ipa user
+        cmd = ['ipa', 'user-add',
+               '--first', username,
+               '--last', username,
+               '--password', username]
+        input_passwd = 'Secret123\nSecret123\n'
+        cmd_output = self.master.run_command(cmd, stdin_text=input_passwd)
+        assert 'Added user "%s"' % username in cmd_output.stdout_text
+
+        # kinit with newly created user
+        input_passwd = 'Secret123\nSecret123\nSecret123\n'
+        cmd_output = self.client.run_command(['kinit', username],
+                                             stdin_text=input_passwd)
+
+        # list Kerberos tickets
```

Comment:
Instead of doing a ssh using your krb tkt, perform a ssh from the test controller using login + password. This will also reproduce the issue or show the success of the patch. You can use 
```
        client = paramiko.SSHClient()
        client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
        client.connect(self.master.hostname,
                       username=username,
                       password='Secret123')
        client.close()
```

Reply:
I have changed it to use SSHClient